src/content_sao.cpp

```diff
@@ -289,6 +288,120 @@ void UnitSAO::notifyObjectPropertiesModified()
 	m_properties_sent = false;
 }
 
+std::string UnitSAO::generateUpdateAttachmentCommand() const
+{
+	std::ostringstream os(std::ios::binary);
+	// command
+	writeU8(os, AO_CMD_ATTACH_TO);
+	// parameters
+	writeS16(os, m_attachment_parent_id);
+	os << serializeString(m_attachment_bone);
+	writeV3F32(os, m_attachment_position);
+	writeV3F32(os, m_attachment_rotation);
+	return os.str();
+}
+
+std::string UnitSAO::generateUpdateBonePositionCommand(const std::string &bone,
+	const v3f &position, const v3f &rotation)
```

Comment:
is there a reason this isn't a const method?

Reply:
it's static, it cannot be const because there is no membership